src/libpriv/rpmostree-core.c

```diff
@@ -1998,6 +2331,23 @@ checksum_version (GVariant *checksum)
   return g_strdup (ret);
 }
 
+static guint
+pkg_hash (gconstpointer v)
+{
+  DnfPackage *p = (DnfPackage*)v;
+  return g_str_hash (dnf_package_get_package_id (p));
+}
+
+static gboolean
+pkg_equal (gconstpointer v1,
+           gconstpointer v2)
+{
+  DnfPackage *p1 = (DnfPackage*)v1;
+  DnfPackage *p2 = (DnfPackage*)v2;
+  return g_str_equal (dnf_package_get_package_id (p1),
+                      dnf_package_get_package_id (p2));
```

Comment:
Hmm, though that only compares the NEVRA, whereas `get_package_id` includes repo information. E.g. a package provided locally should still be considered distinct from a repo package, even if they have the same NEVRA. I think it wouldn't make a difference in this context, since by the time we add to that array, we've already made our determination, though it's a nice property to keep I think (and it's cached too, thankfully).